cdap-api-common/src/main/java/io/cdap/cdap/api/data/format/StructuredRecord.java

```diff
@@ -367,8 +367,41 @@ private Builder(Schema schema) {
      * @throws UnexpectedFormatException if the field is not in the schema, or the field is not nullable but a null
      *                                   value is given
      */
+
+    private boolean checkAndSetLogicalSchema(Schema schema, String fieldName, @Nullable Object value) {
+      schema = getLogicalTypeSchema(schema, EnumSet.of(LogicalType.FIXED));
+      if (schema == null) {
+        return false;
+      }
+      if (schema.getLogicalType().equals(LogicalType.FIXED)) {
```

Comment:
`schema` will always be `Fixed` because the above check is ensuring this. As long as `schema` is not-null, you can call the function to set the fixed type schema and return true.

Reply:
Let us have this fixed logical type check to be specific.If we want to remove specific logic type functionality and merge with set() this check can be useful.